src/storage/src/sink/kafka.rs

```diff
@@ -214,28 +215,19 @@ impl KafkaSinkSendRetryManager {
 
 pub struct SinkProducerContext {
     metrics: Arc<SinkMetrics>,
+    status_tx: mpsc::Sender<SinkStatus>,
     retry_manager: Arc<Mutex<KafkaSinkSendRetryManager>>,
 }
 
-impl SinkProducerContext {
-    pub fn new(
-        metrics: Arc<SinkMetrics>,
-        retry_manager: Arc<Mutex<KafkaSinkSendRetryManager>>,
-    ) -> Self {
-        SinkProducerContext {
-            metrics,
-            retry_manager,
-        }
-    }
-}
-
 impl ClientContext for SinkProducerContext {
     // The shape of the rdkafka *Context traits require us to forward to the `MzClientContext`
     // implementation.
     fn log(&self, level: rdkafka::config::RDKafkaLogLevel, fac: &str, log_message: &str) {
         MzClientContext.log(level, fac, log_message)
     }
-    fn error(&self, error: rdkafka::error::KafkaError, reason: &str) {
+    fn error(&self, error: KafkaError, reason: &str) {
+        let status = SinkStatus::Stalled(error.to_string());
+        let _ = self.status_tx.try_send(status);
```

Comment:
I am not sure the `try_send` here is the right option; we are probably fine with a long enough channel, but we should revisit this in the future!

Reply:
My thesis was:
- If we're getting a ton of errors in a row, it's not that meaningful which error "wins".
- I never want to block the thread that belongs to rdkafka.

So I think it's a good idea to drop messages instead of blocking here or otherwise trying to signal the error. I would have liked even more to have kept the more recent errors instead of the older ones... but there was no convenient library for that as far as I could tell, and it didn't seem important enough to invent one.

What are you thinking here?

Reply:
After thinking about this, i also realized that we were completely dropping these errors on the floor before, so this is a huge upgrade from before; if we end up wanting to backpressure rdkafka so we can record ALL errors, we can switch to https://docs.rs/tokio/latest/tokio/sync/mpsc/struct.Sender.html#method.blocking_send

> I would have liked even more to have kept the more recent errors instead of the older ones
https://docs.rs/tokio/latest/tokio/sync/watch/index.html kindof does this, depending on how you use it!